contributor_docs/unit_testing.md

```diff
@@ -139,3 +139,31 @@ If you need to add a new test file, add it to that folder, then add the filename
 When you add a new test, running `npm test` will generate new screenshots for any visual tests that do not yet have them. Those screenshots will then be used as a reference the next time tests run to make sure the sketch looks the same. If a test intentionally needs to look different, you can delete the folder matching the test name in the `test/unit/visual/screenshots` folder, and then re-run `npm test` to generate a new one.
 
 To manually inspect all visual tests, run `grunt yui:dev` to launch a local server, then go to http://127.0.0.1:9001/test/visual.html to see a list of all test cases.
+
+
+The visual test environment is set up to execute your commands sequentially rather than running a preload or draw function that you provide.
+In continuous integration (CI) environments, it's crucial to keep tests running as quickly as possible. Running a full `preload/draw` cycle as in a regular p5.js sketch can significantly slow down the testing process.
+When testing features like 3D model rendering, you might encounter scenarios where you need to load a model before performing assertions. Here's an example of how you can handle Sequential Command Execution and Asynchronous Operations in your visual tests:
```

Comment:
Rather than saying "Sequential Command Execution and Asynchronous Operations", it might be simpler to mention something like, return a promise that resolves when you have finished everything you want to test

Reply:
To address scenarios involving operations like 3D model rendering, consider returning a promise that resolves upon completing all the necessary tests, ensuring efficiency in your visual testing approach.
@davepagurek Does this make more sense? Do we need more modifications? Maybe remove mentioning 3D models?

Reply:
Yes, Thanks that would make it much more clear. @davepagurek